src/WindowTracker.vala

```diff
@@ -63,7 +63,46 @@ public class Gala.WindowTracker : GLib.Object {
             }
         }
 
-        return null;
+        // try to guess app_info name by process name
+        var file = File.new_for_path ("/proc/%d/comm".printf ((int) pid));
+
+        try {
+            var dis = new DataInputStream (file.read ());
+            var name = dis.read_line (null);
+
+            if (name != null && name != "") {
+                name = name.concat (".desktop");
+                unowned var app = AppSystem.get_default ().lookup_app (name);
+                if (app != null) {
+                    return app;
+                }
+            }
+
+            // this particular process doesn't have corresponding desktop file
+            // but maybe parent process has it
+
+            var stat_file = File.new_for_path ("/proc/%d/stat".printf ((int) pid));
+            var stat_dis = new DataInputStream (stat_file.read ());
+            var stats = stat_dis.read_line (null).split (" ");
+
+            if (stats.length < 3) {
+                warning ("WindowTracker.get_app_from_pid: stat file has wrong format");
```

Comment:
Should there be a return here? (And in the next if)
Also you should probably be checking for < 4 here since you acces at index 3

Reply:
Thanks for noticing that. Fixed!